src/ahbicht/expressions/requirement_constraint_expression_evaluation.py

```diff
@@ -85,10 +85,35 @@ def _or_xor_composition(
                 and left.conditions_fulfilled != ConditionFulfilledValue.NEUTRAL
             )
         ):
-            raise NotImplementedError(
-                "Combining a neutral element with a boolean value in an"
-                f"{composition} is not implemented as it has no useful result."
+            neutral_element: ConditionNode
+            boolean_element: ConditionNode
+            if right.conditions_fulfilled == ConditionFulfilledValue.NEUTRAL:
+                neutral_element = right
+                boolean_element = left
+            else:
+                neutral_element = left
+                boolean_element = right
+            error_message = (
+                f"Combining a neutral element '{neutral_element}' with a boolean value {boolean_element} "
+                f"in an {composition} is not implemented as it has no useful result."
             )
+            if isinstance(neutral_element, Hint):
+                # This error is quite popular in AHBs.
+                # The BDEW tends to write stuff like "[123] O [584]" where [123] is a regular RC and [584] is
+                # something like "Verwendung der ID der Markt/Messlokation". Their intention is that the hint on the
+                # right side should be evaluated as if it was a self-referencing Requirement Constraint that
+                # magically evaluates to Fulfilled/Unfulfilled instead of being treated as what it is: A plain hint.
+                # This is similar to what they tried with [93x] (german datetime) format constraints.
+                # But other than the 931-935 FCs this kind of expression is not only a bad idea but also not even
+                # valid according to their own set of rules.
+                # If you encounter such an error in the AHBs, please message the EDI@Energy Arbeitsgruppe to fix it.
+                # The easiest workaround (beyond ignoring it) is probably to convert it to an _xor_ expression:
+                # Instead of "[123] O [584]" write "[984][123] X [983][1]" where
+                # - [1] is an artificial requirement constraint, that should always evaluate to FULFILLED
```

Comment:
Gibt es diese [1] Bedingung schon?

Reply:
89e077e831313d346fd22a3165766b7884575253 <-- macht es das klarer?

Reply:
Wobei es evtl. eine Überlegung wert wäre, eine Bedingung die immer erfüllt ist und eine die nie erfüllt ist allgemein genau für sowas anzulegen, damit das nicht jeder selbst macht und wir die Nummern auch gut erkennen im bugfixing. 
Muss aber auf jeden Fall nicht in diesen PR ;)

Reply:
kennst du spontan eine nummer, die nie verwendet wird? sonst könnten wir den parser aufbohren, dass er auch sowas wie "[Fulfilled] U [123]" versteht?

Reply:
Ich glaube [1] und [2] gibt es tatsächlich nicht, müsste man aber nochmal überprüfen. Aber [Fulfilled] / [Unfulfilled] wäre natürlich auch sehr elegant und eindeutig.